### PR TITLE
[FIX] core: unexpected display_name "False"

### PR DESCRIPTION
**Pull request by rco-odoo**
If a form view contains the field `display_name`, when creating a new record, the initial value of `display_name` is the string `"False"`, and that weird value also appears in the breadcrumb instead of `"New"`.  In order to avoid this unexpected behavior, the conversion of a value to a display name should be `False`, like any other field would.

### CODE REVIEW

**Review by aab-odoo — APPROVED** [review]
:+1:

**Comment by rco-odoo** [comment]
@robodoo r+

**Comment by qsm-odoo** [comment]
Hello @rco-odoo @aab-odoo 

There is a bug with blog posts since this PR. I investigated a bit but before fixing the symptoms on our side maybe this will require some changes in the ORM first so I'd like your opinion about this:

Before this PR:

After this PR:


Note:
- Arguably strange to allow `''` as value for a required field (but I guess that has always been the webclient which enforces that a required field does not receive the empty string as value)
- The `name` field definition of blog posts is very strange (to be kind): `required=True, default=''` (FP wanted to keep the possibility to create posts without title from the frontend but I did not realize they actually were declared required :grimacing:)

The bug we see in blog:
- Create a blog post from the frontend, leave the title empty and save
- Go to the /blog page => crash

Thanks in advance :)

**Comment by rco-odoo** [comment]
@qsm-odoo does https://github.com/odoo/odoo/pull/86567 make a difference?

Can you be more precise than "bug" and "crash"?  What is the error?

**Comment by qsm-odoo** [comment]
@rco-odoo I reported the issue after the PR you mentioned was merged so I don't think it makes a difference. I retested to be sure on a rebased master of now and still found the result I find inconsistent:
```
In [1]: post = env['blog.post'].create({'blog_id': 1})

In [2]: post.name
Out[2]: ''

In [3]: post.display_name
Out[3]: False
```

Like I said I did not go further in investigating the symptoms of that inconsistency (the crash in blog). But for the record, here is the crash:
image

`AttributeError: 'bool' object has no attribute 'replace'` so either in our blog code or in t-field code, there must a `replace` on something we suppose is a string and which now can be `False`. Won't be difficult to fix that symptom, I just wanted your opinion on if I should even care about investigating and fixing that symptom since the inconsistency I found.

So in short: I am not asking to fix the blog post error (I'll do it), just want to be sure it's an error we want to fix :wink:

**Comment by rco-odoo** [comment]
@Gorash Can you have a look at the error above?
It looks like the template evaluates the `t-field` before the `t-if`.
The error occurs on saas-15.2 BTW.

**Comment by rdeodoo** [comment]
Hello @rco-odoo 
It looks like you are kind of missing @qsm-odoo actual request / question.
It has nothing to do with the template or any module, he is just asking if the following is considered ok or a bug / error:

```
In [1]: post = env['blog.post'].create({'blog_id': 1})

In [2]: post.name
Out[2]: ''

In [3]: post.display_name
Out[3]: False
```

Left is 15.0, right is master
image

It should definitely return an empty string as in 15.0, right?

**Comment by Gorash** [comment]
@rco-odoo The `author_avatar` exists but has no `display_name`, when the field wants to display the image the error occurs. The qweb field expects to have a string for `display_name`, you would have to put a ``or ''``.
```py
   ...
            filename = record.display_name
        filename = filename.replace('/', '-').replace('\\', '-').replace('..', '--')
```

**Comment by rdeodoo** [comment]
Can we be serious for just 30 seconds? It has been 18 days since @qsm-odoo is trying to know if the low level change on `display_name` is a wanted behavior (pretty sure it is not and should be restore?). And I just summarized his demand [here](https://github.com/odoo/odoo/pull/81788#issuecomment-1090525079).

image

But we are still talking about fixing the symptoms without even knowing if the root cause is considered as a bug and should / will be fixed.

If display_name change is to be remained, there is most likely other places that will need a fix, that's not the point. The point is to know if this is considered as a bug and will be fixed or not, before even considering fixing symptoms.

That's quite a very big change, for us and the community, **is it or not intended**?

**Comment by rco-odoo** [comment]
@rdeodoo @qsm-odoo I did not expect so many things being dependent on that change.
To me `display_name` simply is a char field, and so it can have the value `False`.
I propose to fix the definition of `display_name` so that it matches `name_get()`, and thus defaults on the empty string when there is no actual value to display.

**Comment by qsm-odoo** [comment]
Solved with https://github.com/odoo/odoo/pull/88381
